src/ert/scheduler/job.py

```diff
@@ -165,7 +166,58 @@ async def _max_runtime_task(self) -> None:
         )
         self.returncode.cancel()
 
+    async def check_manifest(self, timeout: int = 120) -> None:
+        if self._scheduler.manifest is None:
+            return
```

Comment:
It was required because of some tests that are mocking the creation of the runpath and in those cases no manifest file is in the runpath so I either had to modify those tests or add this safeguard here, but I agree it is a bit strange I will refactor and remove it.

Reply:
Refactor to avoid this ugly logic.